### PR TITLE
add missing examples

### PR DESCRIPTION
**Pull request by datner**
and fix the broken link for Option

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -145,15 +145,60 @@ console.log(result) // { _tag: "Some", value: 0 }
...
+  Match.when({ _tag: "Right" }, (_) => _.right),
+  Match.exhastive, // TypeError! { _tag: "left", left: never } is not assignable to never
+)
+
+```
 
 ### `orElse`
 
-**TODO**
+A Matcher that marks the end of the matcher and allows to provide a fallback value if a no matchers hit. Returns the match.
```
Comment (tim-smart) [inline README.md:166]: `provide a fallback value if a no patterns match`

Also same as above regarding `Returns the match`

File: README.md [hunk 2]
```diff
@@ -145,15 +145,60 @@ console.log(result) // { _tag: "Some", value: 0 }
...
+
+console.log(match("hello")) // "I literally do not understand"
+console.log(match("hi")) // "hello"
+```
+
 
 ### `either`
 
-**TODO**
+A Matcher that _might_ match a value. Returns an [Either](https://github.com/fp-ts/core/blob/main/guides/Either.md) in the shape of `Either<OriginalInput, MatchResult>`.
```
Comment (tim-smart) [inline README.md:185]: `Either<NoMatchResult, MatchResult>`

The input type will be narrowed as much as possible.

**Review by tim-smart — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (datner) [reply README.md]: I was having a stroke while writing this apparently

**Review by tim-smart — APPROVED** [review]

**Comment by tim-smart** [comment]
Thanks!

**Pull request merged**